mautic/mautic
# The contact history items are sorted by time (Add support for microseconds to datetime type)

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


Events in the Timeline History tab on the contact page that occur in the same second can be ordered differently and sometimes in the wrong order. This PR adds support for microseconds to the date time type, which allows sorting in a better order.

Campaign:
![image](https://user-images.githubusercontent.com/96085911/229362313-e09e26e9-01c0-45d9-9656-4397fe1337ab.png)

Contact's timeline (current branch):
![image](https://user-images.githubusercontent.com/96085911/229362397-b803f547-d98a-468b-9f54-1c307cfba1f4.png)

Contact's timeline (5.x):
![image](https://user-images.githubusercontent.com/96085911/229364228-3133ca76-2952-4373-8346-d93cbf747b3d.png)

 
#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a campaign with two or more actions
3. Trigger the campaign
4. Check an order of events in the Timeline History tab on the contact page
